plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -284,6 +284,18 @@
     protected String _rngPath = "/dev/random";
     protected int _rngRatePeriod = 1000;
     protected int _rngRateBytes = 2048;
+    protected String _agentHooksBasedir = "/etc/cloudstack/agent/hooks";
```

Comment:
@bwsw I did not review the whole PR, is it related or similar to http://docs.cloudstack.apache.org/en/latest/adminguide/hosts.html#kvm-libvirt-hook-script-include ?

Reply:
@rhtyd no it's not exactly what does qemu-hooks:

1. there is no qemu hook to override XML
2. qemu hooks deadlocks in certain cases (for onStart/onStop):

```
A hook script must not call back into libvirt, as the libvirt daemon is already waiting for the script to exit.

A deadlock is likely to occur.
```
From: https://www.libvirt.org/hooks.html
So, libvirt qemu hooks are pretty tough stuff:

Reply:
Initially I tried to use qemu-hooks for my task, but lately moved my code to security groups python code, because, otherwise, it works unstable because of the reasons listed above.